### ISSUE TITLE
Alternative implementation approach

### ISSUE THREAD

**Issue by gasche** [reactiveml/rmlbuild#1]
You shouldn't need to fork the whole codebase, because ocamlbuild provides `ocamlbuild_pack.cmo` that contains all the modules and the logic. So you should be able to just write your `rmlbuild_specific.ml` file (opening Ocamlbuild_pack in front), then fork the `main.ml` file, and then write a tiny `rmlbuild.ml` that just call the `main ()` function, like `ocamlbuild.ml` does:

```
open Ocamlbuild_pack
Ocamlbuild_unix_plugin.setup ();
Rml_main.main ()
```

I think that doing this would minimize synchronization woes for you if the ocamlbuild sources evolve.

**Comment by mandel** [comment]
You are right, we should try to reimplement rmlbuild as you suggest. This code was simply an update of the old rmlbuild adapted to the new implementation of ocamlbuild.
